src/extension/utils.ts

```diff
@@ -310,3 +311,7 @@ export function replaceOutput(
 export function getGrpcHost() {
   return `${SERVER_ADDRESS}:${getPortNumber()}`
 }
+
+export async function isPortAvailable(port: number): Promise<boolean> {
+  return (await getPort({ port })) === port
```

Comment:
I'm wondering if we should restrict this a little bit, by specifying a range of ports that are desirable to use, starting with **7863**, also to reduce the chances that after the first iteration it will find a port being used.

```
import getPort, {portNumbers} from 'get-port';

await getPort({port: portNumbers(7863, 8000)})
```

Reply:
Having a ceiling on the port number is not mutually exclusive with the incremental approach. In a small follow-up PR we could add that. However, I'd suggest providing for more headroom than 8000.

Reply:
@sourishkrout Ah I see. The code as provided though will assign randomly in the range. 

If the port exceeds the ceiling, would you basically just error at that point?

Reply:
Oh, sorry... didn't see at first that the API implemented that. Predictable ports will make examples of interop between extension and CLI easier to demo until we figure out "service discovery." That's the main motivation why I'm attached to the incremental assignment approach.... for now.

Reply:
> If the port exceeds the ceiling, is the idea that you would error at that point?

Yep.